### PR TITLE
Added possibility to force collision avoidance processing

### PR DESCRIPTION
**Pull request by AndreaCatania**
Added possibility to force collision avoidance processing

This work has been kindly sponsored by IMVU.

### CODE REVIEW

File: doc/classes/Navigation2DServer.xml [hunk 1]
```diff
@@ -139,6 +139,17 @@
...
 		</method>
+		<method name="agent_force_process_avoidance" qualifiers="const">
+			<return type="Vector2">
+			</return>
+			<argument index="0" name="agent" type="RID">
+			</argument>
+			<argument index="1" name="delta" type="float">
+			</argument>
+			<description>
+				Performs collision avoidance immediately for the specified agent. Please use this function as last solution because it's not optimal.
```
Comment (Calinou) [inline doc/classes/Navigation2DServer.xml]:
```suggestion
				Performs collision avoidance immediately for the specified agent. Only use this method as a last resort, as it's relatively slow.
```

File: doc/classes/NavigationAgent.xml [hunk 2]
```diff
@@ -106,6 +106,15 @@
 				Sends the passed in velocity to the collision avoidance algorithm. It will adjust the velocity to avoid collisions. Once the adjustment to the velocity is complete, it will emit the [signal velocity_computed] signal.
 			</description>
 		</method>
+		<method name="force_process_avoidance">          
+			<return type="Vector3">
+			</return>
+			<argument index="1" name="delta" type="float">
+			</argument>
+			<description>
+				Performs collision avoidance immediately for this agent. Please use this function     as last solution because it's not optimal.
```
Comment (YeldhamDev) [inline doc/classes/NavigationAgent.xml]: Lots of extra spaces here.

File: doc/classes/NavigationAgent2D.xml [hunk 3]
```diff
@@ -106,6 +106,15 @@
 				Sends the passed in velocity to the collision avoidance algorithm. It will adjust the velocity to avoid collisions. Once the adjustment to the velocity is complete, it will emit the [signal velocity_computed] signal.
 			</description>
 		</method>
+		<method name="force_process_avoidance">                                 
+			<return type="Vector2">
+			</return>
+			<argument index="1" name="delta" type="float">
+			</argument>
+			<description>
+				Performs collision avoidance immediately for this agent. Please use this function     as last solution because it's not optimal.
```
Comment (YeldhamDev) [inline doc/classes/NavigationAgent2D.xml]: And here.

File: doc/classes/NavigationServer.xml [hunk 4]
```diff
@@ -139,6 +139,17 @@
...
 		</method>
+		<method name="agent_force_process_avoidance" qualifiers="const">
+         		<return type="Vector3">
+         		</return>
+         		<argument index="0" name="agent" type="RID">
+         		</argument>
+         		<argument index="1" name="delta" type="float">
+         		</argument>
+         		<description>
+				Performs collision avoidance immediately for the specified agent. Please use this function     as last solution because it's not optimal.
```
Comment (YeldhamDev) [inline doc/classes/NavigationServer.xml]: Here as well.

**Comment by asheraryam** [comment]
If I understand correctly, this would be used to force the collision avoidance to happen earlier so we don't have to wait for the next `process` iteration?

**Comment by AndreaCatania** [comment]
With the current workflow the event is emitted in the same frame of when you set the velocity.

So, The current method is always the best solution.

This function can be used to obtain the safe velocity immediately, and got added for some particular cases when it's not really possible wait for the computation to happen naturally.

This function is not optimal and you have to avoid use it.

Regarding the example, you simply need to call the method.

**Comment by smix8** [comment]
So is this still desired or what are we going to do with this pr?
Merge or discard to the salvage yard?

**Comment by AndreaCatania** [comment]
Super seeded by https://github.com/godotengine/godot/pull/66454

**Comment by AndreaCatania** [comment]
Superseded by https://github.com/godotengine/godot/pull/66454

**Pull request closed without merging**